src/Transports.AspNetCore/WebSockets/BaseSubscriptionServer.cs

```diff
@@ -0,0 +1,502 @@
+using System.Security.Claims;
+using System.Security.Principal;
+using GraphQL.Execution;
+using GraphQL.Transport;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// Manages a WebSocket message stream.
+/// </summary>
+public abstract partial class BaseSubscriptionServer : IOperationMessageProcessor
+{
+    private volatile int _initialized;
```

Comment:
```suggestion
    private int _initialized;
```

volatile is not required for Interlocked API.

Reply:
> volatile is not required for Interlocked API.

It also makes calls to `Thread.VolatileRead` unnecessary.